packages/cli/src/databases/db-connection.ts

```diff
@@ -0,0 +1,100 @@
+import { Memoized } from '@n8n/decorators';
+import { Container, Service } from '@n8n/di';
+import { DataSource } from '@n8n/typeorm';
+import { ErrorReporter } from 'n8n-core';
```

Comment:
`@n8n/db` can't depend on `core`, `ErrorReporter` will have to move to `@n8n/backend-common` at some point

Reply:
I wasn't planning to move this particular file. I think this should stay in `cli`

Reply:
I expect `@n8n/integration-test-utils` will need access to DB connection options to set up the test DB that will be used by the utils consumed by `cli` and extensions.